framework/OutStreams/PlotInterfaces/PopulationPlot.py

```diff
@@ -0,0 +1,166 @@
+        descr=r"""Names of the variables from the DataObject to be plotted on a log scale."""))
+    spec.addSub(InputData.parameterInputFactory('index', contentType=InputTypes.StringType,
+        descr=r"""Names of the variable that refers to the batch index"""))
+    spec.addSub(InputData.parameterInputFactory('how', contentType=InputTypes.StringType,
+        descr=r"""Digital format of the generated picture"""))
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag   = 'GAPopulation Plot'
+    self.source     = None      # reference to DataObject source
+    self.sourceName = None      # name of DataObject source
+    self.vars       = None      # variables to plot
+    self.logVars    = None      # variables to plot in log scale
+    self.index      = None      # index ID for each batch
+    self.how        = None      # format of the generated picture
+
+  def handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super().handleInput(spec)
+    params, notFound = spec.findNodesAndExtractValues(['source','vars','index','how'])
+
+    for node in notFound:
+      self.raiseAnError(IOError, "Missing " +str(node) +" node in the PopulationPlot " + str(self.name))
+    else:
+      self.sourceName = params['source']
+      self.vars       = params['vars']
+      self.index      = params['index']
+      self.how        = params['how']
+
+    params, notFound = spec.findNodesAndExtractValues(['logVars'])
+    if notFound:
+      self.logVars = None
+    else:
+      self.logVars = params['logVars']
+
+
+  def initialize(self, stepEntities):
+    """
+      Function to initialize the OutStream. It basically looks for the "data"
+      object and links it to the system.
+      @ In, stepEntities, dict, contains all the Objects are going to be used in the
+                                current step. The sources are searched into this.
+      @ Out, None
+    """
+    src = self.findSource(self.sourceName, stepEntities)
+    if src is None:
+      self.raiseAnError(IOError, f'No source named "{self.sourceName}" was found in the Step for SamplePlot "{self.name}"!')
+    self.source = src
+
+    dataVars = self.source.getVars()
+    missing = [var for var in (self.vars) if var not in dataVars]
+    if missing:
+      msg = f'Source DataObject "{self.source.name}" is missing the following variables ' +\
+            f'expected by OptPath plotter "{self.name}": '
+      msg += ', '.join(f'"{m}"' for m in missing)
+      self.raiseAnError(IOError, msg)
+
+  def run(self):
+    """
+      Main run method.
+      @ In, None
+      @ Out, None
+    """
+    data = self.source.asDataset().to_dataframe()
+    inVars  = self.source._inputs
+    outVars = self.source._outputs
```

Comment:
can you use "getVars()" function to retrieve the inputs and outputs?

Reply:
resolved with getVars